aten/src/ATen/native/Copy.cpp

```diff
@@ -8,52 +8,81 @@
 
 namespace {
 
-template <typename self_T, typename src_T>
-void _copy__cpu(at::Tensor& self, const at::Tensor& src) {
-  at::CPU_tensor_apply2<self_T, src_T>(
-      self, src, [](self_T& self_val, const src_T& src_val) {
-        self_val = static_cast<self_T>(
-            static_cast<at::native::inter_copy_type_t<self_T>>(src_val));
+template <typename dst_T, typename src_T>
+void copy_pointwise_(at::Tensor& dst, const at::Tensor& src) {
+#ifdef _OPENMP
+  if (at::in_parallel_region()) {
+    at::CPU_tensor_parallel_apply2<dst_T, src_T>(
+        dst, src, [](dst_T& dst_val, const src_T& src_val) {
+          dst_val = static_cast<dst_T>(
+              static_cast<at::native::inter_copy_type_t<dst_T>>(src_val));
+        });
+    return;
+  }
+#endif
+  at::CPU_tensor_apply2<dst_T, src_T>(
+      dst, src, [](dst_T& dst_val, const src_T& src_val) {
+        dst_val = static_cast<dst_T>(
+            static_cast<at::native::inter_copy_type_t<dst_T>>(src_val));
       });
 }
 
-template <typename self_T>
-void _copy__cpu(at::Tensor& self, const at::Tensor& src) {
-  AT_CHECK(self.numel() == src.numel(), "sizes do not match");
-  AT_DISPATCH_ALL_TYPES_AND_HALF(
-      src.type(), "_copy__cpu", [&]() { _copy__cpu<self_T, scalar_t>(self, src); });
+template <typename dst_T>
+void copy_pointwise_(at::Tensor& dst, const at::Tensor& src) {
+  AT_DISPATCH_ALL_TYPES_AND_HALF(src.type(), "copy_pointwise", [&]() {
+    copy_pointwise_<dst_T, scalar_t>(dst, src);
+  });
 }
 
-bool copy_transpose_valid(const at::Tensor& self, const at::Tensor& src) {
-  const int MIN_SZ = 60 * 60;
-  return self.is_contiguous() && src.numel() != 0 && src.dim() == 2 &&
-      src.stride(0) == 1 && src.stride(1) == src.size(0) &&
-      self.numel() >= MIN_SZ;
+void copy_pointwise_(at::Tensor& dst, const at::Tensor& src) {
+  AT_DISPATCH_ALL_TYPES_AND_HALF(dst.type(), "copy_pointwise", [&]() {
+    copy_pointwise_<scalar_t>(dst, src);
+  });
 }
 
-} // namespace
+template <typename dst_T, typename src_T>
+void copy_vectorize_(at::Tensor& dst, const at::Tensor& src) {
+  dst_T* dst_ptr = dst.data<dst_T>();
+  src_T* src_ptr = src.data<src_T>();
 
-namespace at {
-namespace native {
+  auto sample = [&](int64_t begin, int64_t end) {
+    int64_t len = end - begin;
+    dst_T* dst_seg = dst_ptr + begin;
+    src_T* src_seg = src_ptr + begin;
+    at::vec256::convert<src_T, dst_T>(src_seg, dst_seg, len);
```

Comment:
You need to write kernels under `native/cpu` for them to be vectorized. Using `vec256` outside that folder won't vectorize your code.